### DISCUSSION TITLE
[Question][Hw1] Clarifying the semantics of arithmetic and comparison operators

### DISCUSSION THREAD

**Discussion by retroinspect** [prosyslab-classroom/cs424-program-reasoning#392] [Q&A]
![image](https://user-images.githubusercontent.com/48230029/190682911-5a143fd7-5d07-4b57-8411-d9db8cd34316.png)

According to homework.pdf, semantics of expression are defined as above.
Can I assume that E1 and E2 are limited to pure values (n / true / false) or labels?
As far as I understand, LLVM never generates nested instructions. 

p.s.  I am asking this question in case of nested instruction exists and LLVM considers it as 'integer' or 'boolean' as label do)

**Comment by sujin0529** [comment]
Hi,
1) E1 and E2 could be n, true, false and x (variable).
2) Sorry... I'm not understanding that meaning of "nested instruction". Can you explain a little more?

**Comment by retroinspect** [comment]
One example of nested instruction would be: ` %div = sdiv i32 3, (sdiv i32 6 3)`
If this kind of instruction is allowed, E1 and E2 would be not limited to n, true, false, and variable.

**Comment by Hhro** [comment]
Hi,
In short, you don’t need to consider the ‘nested instruction’ you mentioned above.
Thanks.
